docs/reference_guides/state_space.rst

```diff
@@ -3,156 +3,187 @@ The State Space
 
 .. currentmodule:: respy.state_space
 
-The implementation of the state space in respy allows the user to solve and analyze a
-wide range of models in an efficient way by storing the essential information about the
-model and by acting as a precise and simple interface between different components of
-the model. First and foremost a state space contains a register all possible states of
-the universe that a particular models allows for. Once a model contains a rich set of
-features it is vital to not repeat information such as to keep the analysis efficient.
-respy defines full states implicitly as combinations of more coarse states to avoid any
-duplication. Furthermore we group states in a way such that all members of one group are
-treated symmetrically throughout the model solution. The state space moreover contains a
-set of methods that facilitate required communications between different state space
-groups. This guide contains an explanation the most important components of the state
-space.
-
+The implementation of the state space in respy serves several purposes.
```

Comment:
I think this paragraph should be split into two parts.

The first part defines what is meant by the state space which is exactly what you do in the beginning. If you feel you delve too much into the economics, you could draft references to the explanations written by Rafael and Benedikt.

The other part defines the problem we need to solve. Talk about memory demands. Reference the curse of the dimensionality. Maybe a back-of-the-envelop calculation: 52 Mio states for KW97. A float64 variables costs 400mb and int8 52mb. The other part defines the problem we need to solve. Talk about memory demands. Reference the curse of the dimensionality. Maybe a back-of-the-envelop calculation: 52 Mio states for KW97. A float64 variables costs 400mb and int8 52mb. What is a float? 10 wages and nonpecs, 1 expected vlaue functions, temporariliy 5 continuation values, 4 of 16 coviarates-> 20 * 400mb = 8GB. What is uint8? 6 state dimensions, 12 covariates -> 18 * 52MB = 1GB.

Then, I could also add some history information. The evolution of the state space: storing all states in tabular format and having a matrix to find child states -> having core states in tabular format and adding dense information if necessary combined with period specific indexer.